### PR TITLE
`scripts/bootstrap` to rule them all

### PR DESCRIPTION
**Pull request by abitrolly**
`make` is not available by default in Fedora containers, so use `scripts/bootstrap` to get all the stuff ready.

Pull Request Checklist

* [x] implement the feature
* [x] write the documentation
* [ ] extend the test coverage
* [ ] update the specification
* [ ] adjust plugin docstring
* [ ] modify the json schema
* [ ] mention the version
* [ ] include a release note

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -65,10 +65,8 @@ rpm: tarball ver2spec  ## Build RPMs
 srpm: tarball ver2spec  ## Build SRPM
 	rpmbuild --define '_topdir $(TMP)' -bs tmt.spec
 
-_deps:  # Minimal dependencies (common for 'deps' and 'develop' targets)
```
Comment (happz) [inline Makefile:68]: This target should be running the newly added script, and both targets bellow would still depend on this target to do the job rather than running it on their own. If there is a "install minimal dependencies" action, it should be a dependable target.

Comment (abitrolly) [reply Makefile:68]: @happz got back the `_deps` target and kept it really minimal.

**Review by martinhoyer — COMMENTED** [review]
Looks good to me. I don't know much about makefiles (and am afk), so can't make a proper review, but if this makes `make` unneeded - yay.

**Comment by LecrisUT** [comment]
Not sure what the state of this would be. I think it would be more helpful to understand what is the end-goal here. E.g. is this meant to prepare a minimum environment, in which case, I would say this is not needed and the developer should use whatever they prefer `pip install -e`, `uv`, etc. Is it supposed to be a complete environment, in which case it should do a `dnf install tmt+all` and override the non-python dependencies with a venv.

I would personally prefer instead to provide developer containers and setup something like `.devcontainer` to advertise that workflow. This would allow us to specify more explicitly what environment we are setting up for the developer.

**Comment by abitrolly** [comment]
@LecrisUT the end goal is to quickly run the new project in isolation to see how it works and if it could be patched. Without spending like hours on browsing documentation.

`.devcontainer` is strictly VS Code thing, no? I don't know how it works and what kind of isolation it provides.

**Comment by LecrisUT** [comment]
> @LecrisUT the end goal is to quickly run the new project in isolation to see how it works and if it could be patched. Without spending like hours on browsing documentation.

Unfortunately the bootstrap script would go against such isolation. Seems like a first-party container would be better?

> `.devcontainer` is strictly VS Code thing, no? I don't know how it works and what kind of isolation it provides.

Jetbrains also recognizes it (though it didn't play well with podman as late as last year when I tested it). But it's basically just a fancy wrapper around containerfiles that some ides recognize for bootstrapping development environments.

**Comment by abitrolly** [comment]
> Unfortunately the bootstrap script would go against such isolation. Seems like a first-party container would be better?

Which container? `podman`, `docker`, `firecracker`, `kubernetes`... At least I know how to run stuff in my container.

**Comment by LecrisUT** [comment]
Sorry for the long time between responses (feel free to ping me any time if it seems we are not getting back to you in a timely manner). Let's get back to the original issue:
1. The format of `scripts/bootstrap` is initially not very appealing because figuring out what needs to be there is complicated. For example we do not strictly depend on `hatch` (we do not use it in the test-suite), while some of the python dependencies are not installable from PyPI (some are installed from sdist and require `krb-devel`). So what would the developer need in the bootstrapping of the project?
2. Do you have some more context on how these "scripts to rule them all" is integrated? Do some IDEs integrate with it, or do users have to be aware that the project is using this, navigate to the folder and run the scripts?
3. The minimal setup does not need any bootstrapping. A simple `pip install -e .` as per any python project is enough to get you started
   - Using the test suite makes things more complicated, but for the most cases `pip install -e .[tests,docs]` will get you going for the pytest tests, or running the beakerlib tests in `tests.sh` will also work (hmm, yes there is the `beakerlib` dependencies that would need to be installed before)
   - More complex tests that require more system packages, those are hard, and the only bootstrap I've found to work for me is to: `dnf install tmt+all`, create a venv with system-site-packages, and `pip install -e .` on top of it
4. A more reliable developer environment *may* be provided with something like devcontainer only because it fixes the OS and we can re-define the dependencies in a consistent manner. We do not provide a developer container now though (we do provide user containers though [see quay.io](https://quay.io/organization/teemtee))
5. If I understand the origin on these is to simplify the onboarding, and :+1: we definitely need one, and I know quite a few projects that would need some help on that as well. What tools are available to do so though. Even if it's just one-liner in the Readme or equivalent where the (drive-by) contributor would expect it and we should add, that would also be something to explore.
6. Finally, things are still constantly moving, e.g. recently the hatch scripts are becoming incompatible because it would not follow the `uv.lock`/`pylock.toml`.

So basically, we are coming back to the onboarding issues for this and trying to figure out what is the best (or not even that) way forward with this. A shell script (and even the current Makefile) is not ideal because it is not very distro agnostic way to work, but if it's the simplest way forward to bridge the gap, let's check if there are not some different contents that should be in there.

**Comment by psss** [comment]
Notes from the triage meeting: Seems that providing a container image with everything installed and prepared could be the easiest way to help users with quick experimenting and the first steps. Tracking issue: #4603.

**Comment by LecrisUT** [comment]
From triage meeting, we are closing this issue for now, but feel free to comment and/or ping us if you need more discussion on it.
